### ISSUE TITLE
Time error loading SSM maintenance windows

### ISSUE THREAD

**Issue by ChrisHoodAU** [awslabs/aws-instance-scheduler#185]
I have the scheduler configured with several SSM maintenance windows, automatic startup during those windows is no longer working.

Scheduler default time zone is set to Australia/Brisbane, as was the maintenance window time zone. The following example error is logged by the EC2 scheduler at every invocation:

`ERROR : Error loading ssm maintenace windows, (time data '2020-07-26T00:00+10:00' does not match format '%Y-%m-%dT%H:%MZ')`

The schedules are then not found by the scheduler:

`SSM maintenance window <window> used in schedule <schedule> not found or disabled`

I've changed the Maintenance window time zone to UTC but this has not solved the issue. Any advice appreciated.

(Minor) Also note the typo in the error message - misspelling of 'maintenance'

**Comment by praveenis** [comment]
Hi @ChrisHoodAU , currently scheduler supports SSM maintenance window with UTC format. we will fix this issue in next release. Changing the SSM maintenance window time zone to UTC should solve the issue. Can you please retry and share the error logs, dynamo DB period and schedule details and maintenance window details. 

Thanks,
Praveen

**Comment by JCKrause** [comment]
I have tested this in our DEV environment and setting the SSM Maintenance Window to UTC does make the error go away and makes the instance scheduler happy.

However setting the SSM Maintenance Windows to UTC presents another issue and a lot of admin overhead.

In order to deploy the instance scheduler to production I will have to change all of the existing SSM Maintenance Window settings from US/Eastern to UTC.

This is okay for a few months until the US time changes again. Then I will have to adjust all the SSM Maintenance Window settings to compensate for the 1 hour time change in order to keep our maintenance on the same schedule year round.

Do you have any idea of when this issue might be addressed?

Thanks,
John

**Comment by andycattle** [comment]
Hi There,

We are getting a similar error, however in our case we are using UTC.

In this case the maintenance window was set to `rate(1 hour)`, seems to expect there to be no seconds or ms in in the time.

```
2020-11-02 - 16:35:47.854 - ERROR   : Error loading ssm maintenace windows, (time data '2020-11-02T17:17:52.856Z' does not match format '%Y-%m-%dT%H:%MZ')
```

Seem this code is at fault in `ec2_service.py`:
```python
execution_time = datetime.strptime(window['NextExecutionTime'], "%Y-%m-%dT%H:%MZ")
```

I think strptime has a tough time with ISO date formats

Thanks,
Andy

**Comment by rgoltz** [comment]
Hi guys,

we hit the same issue here - Since we are working in Timezone: Europe/Berlin. We can confirm the issue reported here and we can add following observation:

We have a bunch of maintenance window set up. Even tough the maintenance window defined in our schedule is UTC, it seems that the error is occuring if **only one** is non-UTC timezone (here in this case `ScheduleTimezone: Europe/Berlin`). To be more specific, this maintenance window in non-UTC timezone format is not used in any of our schedules.

We getting following error from instance scheduler:

> 2021-01-07T16:55:43.735+01:00	2021-01-07 - 15:55:43.735 - ERROR : Error loading ssm maintenace windows, (time data '2021-01-13T03:30+01:00' does not match format '%Y-%m-%dT%H:%MZ')

@praveenis @maykays - Could you please give us an ETA on this fix, since we are not able to change all of our maintenance window configurations - other internal processes depending on the "real" timezone, like Europe/Berlin.

**Comment by maykays** [comment]
This is a known issue. We plan to address this issue in the next release late Q1-early Q2 2021.

**Comment by rgoltz** [comment]
@mobri2a / @maykays - Is the fix still in your task-backlog or it's fixed via recently releaed version[ v1.4.0](https://github.com/awslabs/aws-instance-scheduler/releases/tag/v1.4.0)?
Thanks for any update/input on this :)

**Comment by gockle** [comment]
@rgoltz Yes this issue has been resolved in the v1.4.0.
